src/utils.ts

```diff
@@ -250,3 +303,146 @@ export async function removeFilesAndFolders(workspacename: string, filesAndFolde
+		console.log('Didact configuration updated');
+	}
+}
+
+export async function addNewTutorialWithNameAndCategoryForDidactUri(uri: Uri, name? : string, category? : string) : Promise<void> {
+	let tutorialName = undefined;
+	let tutorialCategory = undefined;
+	if (name && category) {
+		tutorialName = name; 
+		tutorialCategory = category;
+	} else {
+		const categoriesForValidation : string[] = getDidactCategories();
+		const tutorialsForValidation : string[] = getDidactTutorials();
+
+		tutorialName = await getTutorialName(tutorialsForValidation);
+		const selectedCategory : string[] = await quickPickCategory(categoriesForValidation);
+		if (selectedCategory === undefined) {
+			throw Error("Canceled out of Tutorial Category selection");
+		}		
+		tutorialCategory = selectedCategory[0];
+	}
+	
+	if (!uri) {
+		uri = await getCurrentFileSelectionPath();
+	}
+	if (tutorialName && tutorialCategory) {
+		await registerTutorialWithArgs(tutorialName, uri.toString(), tutorialCategory);
+	}
+}
+
+async function getTutorialName(tutorialsForValidation : string[] ) : Promise<string|undefined>{
+	const result = await window.showInputBox({
+		value: 'New Tutorial',
+		placeHolder: 'Enter the name for your new tutorial. The name must be unique.',
+		ignoreFocusOut: true,
+		validateInput: (inputVal: string) => {
+			let val = validateTutorialNameInput(inputVal, tutorialsForValidation);
+			return val;
+		}
+	});
+	if (result === undefined) {
+		throw Error("Canceled out of Tutorial Name input box");
+	}
+	return result?.trim();
+}
+
+function validateTutorialNameInput(value: string, tutorialsForValidation : string[] ): string | null {
+	if (typeof value === "string") {
+	  if (value.trim().length === 0) {
+		return "Empty tutorial name is not allowed";
+	  }
+	  if (tutorialsForValidation && tutorialsForValidation.indexOf(value) > -1) {
+		  return "Tutorial with that name already exists. Tutorial names must be unique."
+	  }
+	  return null;
+	}
+  	return `${value} is invalid`;
+}
+
+async function quickPickCategory(
+	categories: string[],
+	canSelectMany: boolean = false,
+	acceptInput: boolean = true): Promise<string[]> {
+	let options = categories.map(tag => ({ label: tag }));
+
+	return new Promise((resolve, _) => {
+		let quickPick = window.createQuickPick();
+		let placeholder = "Select a Tutorial Category.";
+
+		if (acceptInput) {
+			placeholder = "Select existing Category or type a new name. 'Enter' to confirm. 'Escape' to cancel.";
+		}
+
+		quickPick.placeholder = placeholder;
+		quickPick.canSelectMany = canSelectMany;
+		quickPick.items = options;
+		quickPick.ignoreFocusOut = true;
+		let selectedItems: any[] = [];
+
+		if (canSelectMany) {
+			quickPick.onDidChangeSelection((selected) => {
+				selectedItems = selected;
+			});
+		}
+		quickPick.onDidAccept(_ => {
+			if (quickPick.value.trim().length > 0 || selectedItems.length > 0 || quickPick.activeItems.length > 0) {
```

Comment:
there is a trim here but not on the validation. it means that if a user add an existign name with space after, the validation is not triggered but the created value is erasing the previous template

Reply:
@apupier  The trim is in both places... if you look at line 407, the trim is used there during the validation as well.

Reply:
ok, so do not know where is the glitch but the tutirial is created withotu space and validation does not catch extra space
![validationNotForbiddingSameNameExceptSpaceAtEnd](https://user-images.githubusercontent.com/1105127/112836858-6b551200-909b-11eb-821d-ffbcc2569375.gif)

Reply:
found it. different spot (see the `validateTutorialNameInput` method)